File: Client/Scenes/NetComClasses/GameEndMessage.cs

using System;

namespace Client.Scenes.NetComClasses
{
	public class GameEndMessage : AbstractMessage
	{
		public int winnerID { get; set; }
		public int loserID { get; set; }
		public Object statistics;

		public GameEndMessage(object statistics, int winnerId, int loserId)
		{
			this.statistics = statistics;
			winnerID = winnerId;
			loserID = loserId;
			this.type = MessageType.GAME_END;
		}
	}
}

File: KI/KI-Client/src/main/java/messages/GameEndMessage.java
package messages;

import messages.data.MessageType;

import java.util.List;

/**
 * This class represents a game end message.
 */
public class GameEndMessage extends Message {

  public final int winnerID; // client id of winner
  public final int loserID; // client id of loser
  public final List<String> statistics; // statistics of the game (can be empty)

  public GameEndMessage(final int winnerId, final int loserId, final List<String> statistics) {
    super(MessageType.GAME_END);
    this.winnerID = winnerId;
    this.loserID = loserId;
    this.statistics = statistics;
  }
}
